soc/nordic/nrf54h/CMakeLists.txt

```diff
@@ -3,6 +3,7 @@
 
 if(CONFIG_ARM)
   zephyr_library_sources(soc.c)
+  zephyr_library_sources(../common/soc_lrcconf.c)
```

Comment:
Why is it added only for ARM cores?
I think that instead of the above entry, the following should be added in `soc/nordic/common/CMakeLists.txt` (so that nRF92 could also use it):
```
zephyr_library_sources_ifdef(NRF_PLATFORM_HALTIUM soc_lrcconf.c)
```

Reply:
It was correct to only have it for arm cores, or rather for local domains, as global domains don't have an NRF_LRCCONF010 instance and it breaks their builds.